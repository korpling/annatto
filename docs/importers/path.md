# path (importer)

Add file nodes for all files in the imported directory.

## Configuration

###  corpus_name



