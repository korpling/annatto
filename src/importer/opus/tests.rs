use std::path::Path;

use insta::assert_snapshot;

use crate::{importer::opus::ImportOpusLinks, test_util::import_as_graphml_string};

#[test]
fn serialize() {
    let module = ImportOpusLinks::default();
    let serialization = toml::to_string(&module);
    assert!(
        serialization.is_ok(),
        "Serialization failed: {:?}",
        serialization.err()
    );
    assert_snapshot!(serialization.unwrap());
}

#[test]
fn test_generic_xml() {
    let actual = import_as_graphml_string(
        ImportOpusLinks::default(),
        Path::new("tests/data/import/opus/"),
        None,
    )
    .unwrap();

    assert_snapshot!(actual);
}
