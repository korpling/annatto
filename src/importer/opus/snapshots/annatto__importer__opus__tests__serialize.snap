---
source: src/importer/opus/tests.rs
expression: serialization.unwrap()
---
default_name = "align"
default_type = "Pointing"
reverse_type = "Pointing"
